Structure_Engine/Query/TorsionalConstant.cs

```diff
@@ -355,57 +418,152 @@ public static double ITorsionalConstant(this IProfile profile)
+        }
+
+        /***************************************************/
+
+        [Description("Empirical formula used to correct the torsional constant with enhancement from a T-junction. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
         [Input("tw", "Web thickness, assumed to be the stem of the T.", typeof(Length))]
         [Input("tf", "Flange thickness, assumed to be the top of the T.", typeof(Length))]
         [Input("r", "Root radius, assumed to be the same on both sides of the T.", typeof(Length))]
         private static double AlphaTJunction(double tw, double tf, double r)
         {
+            //Equation 27
             return -0.042 + 0.2204 * tw / tf + 0.1355 * r / tf - 0.0865 * (r * tw) / Math.Pow(tf, 2) - 0.0725 * Math.Pow(tw / tf, 2);
         }
 
         /***************************************************/
 
-        [Description("Emperical formula used to correct the torsional constant with enhancement from a L-junction. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
+        [Description("Empirical formula used to correct the torsional constant with enhancement from a L-junction. Taken from 'P385 Design of steel beams in torsion', Appendix B.")]
         [Input("tw", "Web thickness.", typeof(Length))]
         [Input("tf", "Flange thickness.", typeof(Length))]
         [Input("r", "Root radius.", typeof(Length))]
         private static double AlphaLJunction(double tw, double tf, double r)
         {
+            //Equation 29
             return -0.0908 + 0.2621 * tw / tf + 0.1231 * r / tf - 0.0752 * (tw * r) / Math.Pow(tf, 2) - 0.0945 * Math.Pow(tw / tf, 2);
         }
 
         /***************************************************/
+
+        [Description("Empirical formula used to correct the torsional constant with enhancement from a tapered T-junction such as a taper flange I-Section. Taken from Johnston & El Darwish, 1964")]
+        [Input("tw", "Web thickness.", typeof(Length))]
+        [Input("tf", "Mean flange thickness.", typeof(Length))]
+        [Input("r", "Root radius.", typeof(Length))]
+        [Input("t2", "Thickness of flange at intersection with near face of web.", typeof(Length))]
+        [Input("s", "Flange taper slope.", typeof(Ratio))]
+        private static double AlphaTaperTJunction(double tw, double tf, double r, double t2, double s)
+        {
+            if (0.2 > r / t2 || r / t2 > 1.0)
+                Reflection.Compute.RecordWarning("Calculation of alpha term of torsional constant may not be accurate because ratio of root radius to flange thickness is out of the applicable range.");
+
+            if (0.2 > tw / t2 || tw / t2 > 1.0)
+                Reflection.Compute.RecordWarning("Calculation of alpha term of torsional constant may not be accurate because ratio of web thickness to flange thickness is out of the applicable range.");
+
+            double alpha0 = AlphaTJunction(tw, t2, r);
```

Comment:
Think `tf` should be used here

```suggestion
            double alpha0 = AlphaTJunction(tw, tf, r);
```